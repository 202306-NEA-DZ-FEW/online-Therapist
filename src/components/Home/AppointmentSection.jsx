import { teamMembers } from "@/util/library";
import Button from "../elements/Button";
import Link from "next/link";
import { UserAuth } from "@/context/AuthContext";
import { useTranslation } from "next-i18next";

function AppointmentSection() {
    const { t } = useTranslation("common");
    const { user } = UserAuth();
    // Define the paths based on the user's authentication status
    const bookingPath = user ? "/bookAnAppointment" : "/login/";

    return (
        <div className=' p-10 md:pl-10 bg-LightBeige md:pr-10 lg:pl-32 lg:pr-32'>
            <h2 className='text-center font-semibold lg:font-normal lg:leading-tight lg:text-start pt-12 text-2xl md:text-2xl lg:text-5xl uppercase text-black font-atkinson m-0 pl-6 pr-6'>
                {t("booking.heading1")} <br />
                {t("booking.heading2")}
            </h2>
            <div className='ml-24 lg:ml-0 md:p-8 lg:p-0 max-w-2xl lg:max-w-none'>
                <div className='flex flex-wrap text-base font-semibold leading-7 text-white lg:space-x-10 md:space-x-5 space-x-5/2 items-center justify-evenly'>
                    {teamMembers.map((member) => (
                        <div key={member.name} className='m-10 p-10'>
                            <div className='group flip-card relative'>
                                <div className='flip-card-inner flex flex-col items-end'>
                                    <div className='flip-card-front absolute w-32 h-32'>
                                        <img
                                            src={member.image}
                                            alt={member.name}
                                            className='w-full h-full object-cover rounded-full border text-center transform transition-transform duration-300'
                                        />
                                    </div>
                                    <div className='flip-card-back absolute w-32 h-32 bg-white text-center flex flex-col justify-center rounded-full border text-center transform transition-transform duration-300 opacity-0 hover:opacity-100'>
                                        <span className='text-lg font-semibold text-black'>
                                            {member.name}
                                        </span>
                                    </div>
                                </div>
                            </div>
                        </div>
                    ))}
                </div>
            </div>

            <div className=' text-black lg:text-lg md:text-lg text-md text-left mx-auto mt-20 p-4 md:p-8 lg:p-8 max-w-2xl lg:max-w-none'>
                <p>
                    {t("booking.paragraph")}
                    {/* With us, you can get the same
                    professionalism and quality care as you would expect from an
                    in-office therapist, but with the ability to communicate
                    whenever and however you want. */}
                </p>
            </div>

            <div className='flex justify-center m-4 md:m-6 lg:mb-16 '>
                <Link href={bookingPath}>
                    <Button
                        buttonText={t("hero.bookingButton")}
                        buttonSize='fit'
                    />
                </Link>
            </div>
        </div>
    );
}

export default AppointmentSection;
