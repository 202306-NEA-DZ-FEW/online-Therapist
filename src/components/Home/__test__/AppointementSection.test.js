import renderer from "react-test-renderer";

import AppointmentSection from "../AppointmentSection";

it("renders correctly", () => {
    const tree = renderer.create(<AppointmentSection />).toJSON();
    expect(tree).toMatchSnapshot();
});
