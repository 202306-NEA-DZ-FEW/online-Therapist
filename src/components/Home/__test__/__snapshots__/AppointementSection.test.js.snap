// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className=" p-10 md:pl-10 bg-LightBeige md:pr-10 lg:pl-32 lg:pr-32"
>
  <div
    style={
      Object {
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "opacity": 0,
          "transform": "translateY(75px) translateZ(0)",
        }
      }
    >
      <h2
        className="text-center font-semibold lg:font-normal lg:leading-tight lg:text-start pt-12 text-2xl md:text-2xl lg:text-5xl uppercase text-black font-atkinson m-0 pl-6 pr-6"
      >
        booking.heading1
         
        <br />
        booking.heading2
      </h2>
    </div>
  </div>
  <div
    className="ml-24 lg:ml-0 md:p-8 lg:p-0 max-w-2xl lg:max-w-none"
  >
    <div
      className="flex flex-wrap text-base font-semibold leading-7 text-white lg:space-x-10 md:space-x-5 space-x-5/2 items-center justify-evenly"
    >
      <div
        className="m-10 p-10"
      >
        <div
          className="group flip-card relative"
        >
          <div
            className="flip-card-inner flex flex-col items-end"
          >
            <div
              className="flip-card-front absolute w-32 h-32"
            >
              <img
                alt="Maria JR"
                className="w-full h-full object-cover rounded-full border text-center transform transition-transform duration-300"
                src="https://www.talkspace.com/blog/wp-content/uploads/2020/03/psychologist-psychiatrist-vs-therapist-1-800x500.jpg"
              />
            </div>
            <div
              className="flip-card-back absolute w-32 h-32 bg-white text-center flex flex-col justify-center rounded-full border text-center transform transition-transform duration-300 opacity-0 hover:opacity-100"
            >
              <span
                className="text-lg font-semibold text-black"
              >
                Maria JR
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="m-10 p-10"
      >
        <div
          className="group flip-card relative"
        >
          <div
            className="flip-card-inner flex flex-col items-end"
          >
            <div
              className="flip-card-front absolute w-32 h-32"
            >
              <img
                alt="Marris Adikwu"
                className="w-full h-full object-cover rounded-full border text-center transform transition-transform duration-300"
                src="https://www.talkspace.com/blog/wp-content/uploads/2020/06/Marris_Headshot-150x150.jpg"
              />
            </div>
            <div
              className="flip-card-back absolute w-32 h-32 bg-white text-center flex flex-col justify-center rounded-full border text-center transform transition-transform duration-300 opacity-0 hover:opacity-100"
            >
              <span
                className="text-lg font-semibold text-black"
              >
                Marris Adikwu
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="m-10 p-10"
      >
        <div
          className="group flip-card relative"
        >
          <div
            className="flip-card-inner flex flex-col items-end"
          >
            <div
              className="flip-card-front absolute w-32 h-32"
            >
              <img
                alt="Bisma Anwar"
                className="w-full h-full object-cover rounded-full border text-center transform transition-transform duration-300"
                src="https://www.talkspace.com/blog/wp-content/uploads/2020/09/Bisma_Circle-min-1-150x150.png"
              />
            </div>
            <div
              className="flip-card-back absolute w-32 h-32 bg-white text-center flex flex-col justify-center rounded-full border text-center transform transition-transform duration-300 opacity-0 hover:opacity-100"
            >
              <span
                className="text-lg font-semibold text-black"
              >
                Bisma Anwar
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="m-10 p-10"
      >
        <div
          className="group flip-card relative"
        >
          <div
            className="flip-card-inner flex flex-col items-end"
          >
            <div
              className="flip-card-front absolute w-32 h-32"
            >
              <img
                alt="Joseph Rauch"
                className="w-full h-full object-cover rounded-full border text-center transform transition-transform duration-300"
                src="https://www.talkspace.com/blog/wp-content/uploads/2020/01/Joseph-Rauch.png"
              />
            </div>
            <div
              className="flip-card-back absolute w-32 h-32 bg-white text-center flex flex-col justify-center rounded-full border text-center transform transition-transform duration-300 opacity-0 hover:opacity-100"
            >
              <span
                className="text-lg font-semibold text-black"
              >
                Joseph Rauch
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="m-10 p-10"
      >
        <div
          className="group flip-card relative"
        >
          <div
            className="flip-card-inner flex flex-col items-end"
          >
            <div
              className="flip-card-front absolute w-32 h-32"
            >
              <img
                alt="Jor-El Caraballo"
                className="w-full h-full object-cover rounded-full border text-center transform transition-transform duration-300"
                src="https://www.talkspace.com/blog/wp-content/uploads/2020/06/Jor-El-Caraballo-150x150.jpg"
              />
            </div>
            <div
              className="flip-card-back absolute w-32 h-32 bg-white text-center flex flex-col justify-center rounded-full border text-center transform transition-transform duration-300 opacity-0 hover:opacity-100"
            >
              <span
                className="text-lg font-semibold text-black"
              >
                Jor-El Caraballo
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className=" text-black lg:text-lg md:text-lg text-md text-left mx-auto mt-20 p-4 md:p-8 lg:p-8 max-w-2xl lg:max-w-none"
  >
    <div
      style={
        Object {
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "opacity": 0,
            "transform": "translateY(75px) translateZ(0)",
          }
        }
      >
        <p>
          booking.paragraph
        </p>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "opacity": 0,
          "transform": "translateY(75px) translateZ(0)",
        }
      }
    >
      <div
        className="flex justify-center m-4 md:m-6 lg:mb-16 "
      >
        <a
          href="/bookAnAppointment"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="relative inline-flex items-center justify-center px-6 py-3 overflow-hidden font-medium transition duration-300 ease-out border-2 border-Teal rounded-lg shadow-md group text-xl undefined bg-white hover:cursor-pointer"
          >
            <span
              className="absolute inset-0 flex items-center justify-center w-full h-full text-white duration-300 -translate-x-full bg-Teal group-hover:translate-x-0 ease"
            >
              <svg
                className="w-6 h-6 "
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M14 5l7 7m0 0l-7 7m7-7H3"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                />
              </svg>
            </span>
            <span
              className="absolute flex items-center justify-center w-full h-full text-Teal transition-all duration-300 transform group-hover:translate-x-full ease"
            >
              hero.bookingButton
            </span>
            <span
              className="relative invisible"
            >
              $
              hero.bookingButton
            </span>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
